source/game/kart/KartJump.cc

```diff
@@ -0,0 +1,239 @@
+            {{11.0f, 1.5f, 0.9f, 0.0018f}, {14.0f, 1.5f, 0.9f, 0.0006f},
+                    {7.5f, 2.5f, 0.93f, 0.05f}}};
+
+    static constexpr std::array<f32, 3> FINAL_ANGLES = {{360.0f, 720.0f, 180.0f}};
+
+    if (m_variant == BoostRampVariant::SingleFlipTrick) {
+        m_properties = TRICK_PROPERTIES[0];
+        m_finalAngle = FINAL_ANGLES[0];
+    } else if (m_variant == BoostRampVariant::DoubleFlipTrick) {
+        m_properties = TRICK_PROPERTIES[1];
+        m_finalAngle = FINAL_ANGLES[1];
+    } else if (m_type == TrickType::BikeSideStuntTrick) {
+        m_properties = TRICK_PROPERTIES[2];
+        m_finalAngle = FINAL_ANGLES[0];
+    }
+
+    m_angleDelta = m_properties.initialAngleDiff;
+    m_angleDeltaFactorDec = m_properties.angleDiffMulDec;
+    m_angle = 0.0f;
+    m_angleDeltaFactor = 1.0f;
+}
+
+void KartJump::reset() {
+    m_cooldown = 0;
+}
+
+void KartJump::tryStart(const EGG::Vector3f &left) {
+    if (!state()->isTrickStart()) {
+        return;
+    }
+
+    if (m_move->speedRatioCapped() > 0.5f) {
+        m_variant = static_cast<BoostRampVariant>(state()->boostRampType());
+
+        start(left);
+    }
+
+    state()->setTrickStart(false);
+}
+
+void KartJump::calc() {
+    --m_cooldown;
+    m_cooldown = std::max(0, static_cast<s32>(m_cooldown));
+
+    if (state()->isTrickRot()) {
+        calcRot();
+    }
+
+    calcInput();
+}
+
+bool KartJump::someFlagCheck() {
+    return state()->isTrickStart() || state()->isInATrick();
+}
+void KartJump::calcInput() {
+    constexpr s16 TRICK_ALLOW_TIMER = 14;
+
+    System::Trick trick = inputs()->currentState().trick;
+
+    if (!someFlagCheck() && trick != System::Trick::None) {
+        m_nextTrick = trick;
+        m_nextAllowTimer = TRICK_ALLOW_TIMER;
+    }
+
+    u32 airtime = state()->airtime();
+    if (airtime == 0 || m_nextAllowTimer < 1 || airtime > 10 ||
+            (!state()->isTrickable() && state()->boostRampType() < 0) || someFlagCheck()) {
+        --m_nextAllowTimer;
+        m_nextAllowTimer = std::max(0, static_cast<s32>(m_nextAllowTimer));
+    } else {
+        if (airtime > 2) {
+            state()->setTrickStart(true);
+        }
+        if (state()->isRampBoost()) {
+            m_boostRampEnabled = true;
+        }
+    }
+    if (state()->isTouchingGround() && !collide()->isRampBoost()) {
+        m_boostRampEnabled = false;
+    }
+}
+
+void KartJump::end() {
+    if (state()->isTrickRot()) {
+        physics()->composeDecayingRot(m_rot);
+    }
+
+    state()->setInATrick(false);
+    state()->setTrickRot(false);
+    m_boostRampEnabled = false;
+}
+
+void KartJump::setAngle(const EGG::Vector3f &left) {
+    static constexpr std::array<std::array<std::array<f32, 2>, 3>, 3> ANGLE_PROPERTIES = {
+            {{{{{40.0f, 15.0f}}, {{45.0f, 20.0f}}, {{45.0f, 20.0f}}}},
+                    {{{{36.0f, 13.0f}}, {{42.0f, 18.0f}}, {{42.0f, 18.0f}}}},
+                    {{{{32.0f, 11.0f}}, {{39.0f, 16.0f}}, {{16.0f, 1.0f}}}}}};
```

Comment:
Use trailing commas to force clang-format to separate each element onto newlines.

Reply:
Done, I also have changed the inner array to instead of a `struct AngleProperties` in order to match `TrickProperties`.